src/std_extensions/collections.rs

```diff
@@ -126,22 +193,107 @@ fn list_and_elem_type(list_type_def: &TypeDef) -> (Type, Type) {
     );
     (list_type, elem_type)
 }
+
+/// A list operation
+#[derive(Debug, Clone, PartialEq)]
+pub enum ListOp {
+    /// Pop from end of list
```

Comment:
What is the behaviour of the program if the list is empty? (Panic? Error? Doesn't seem to be captured in the signature.)

Reply:
panics, should be clear in spec once we have it